Docs/ProjectSpec.md

```diff
@@ -663,6 +664,21 @@ targets:
         product: SPMUtility
 ```
 
+Depending on multiple products from a package:
+
+```yaml
+packages:
+  FooFeature:
+    path: Packages/FooFeature
+targets:
+  App:
+    dependencies:
+      - package: FooFeature
+        products:
+          - FooDomain
+          - FooUI
```

Comment:
How can we handle if we want to do below case?

- FooDomain: embed = true
- FooUI: embed = false

Reply:
My thinking was that people would use `product` when needing fine-grained control and `products` (with an s) when products are to be added the same way.

```yaml
dependencies:
- package: FooFeature
  product: FooDomain
  embed: true
- package: FooFeature
  product: FooUI
  embed: false
```

Alternatively, products should be objects as shown below while also supporting products as plain strings. If both aren't supported, the benefit of `products` become too little.

So both of these would be valid:

```yaml
dependencies:
- package: FooFeature
  products:
  - name: FooDomain
    embed: true
  - name: FooUI
    embed: false
```

```yaml
dependencies:
- package: FooFeature
  products:
  - FooDomain
  - FooUI
```

Reply:
@simonbs the `embed` property is part of the Dependency type, so the list of objects would break that relationship.
I think having to split out into seperate `product` is fine. Would like to see that perhaps mentioned in the docs under `product` like "Useful if you want to define different linking options per product. "

Reply:
@yonaskolb That's a good idea. I've added it in https://github.com/yonaskolb/XcodeGen/pull/1395/commits/9785749ad19a107de8f4fb1a7184c0bd5c4a0c12.